cmd/kueuectl/app/list/list_clusterqueue.go

```diff
@@ -144,35 +144,56 @@ func (o *ClusterQueueOptions) validActiveFlagOptionProvided() bool {
 
 // Run prints the cluster queues.
 func (o *ClusterQueueOptions) Run(ctx context.Context) error {
-	continueToken := ""
+	var totalCount int
+
+	opts := metav1.ListOptions{
+		LabelSelector: o.LabelSelector,
+		FieldSelector: o.FieldSelector,
+		Limit:         o.Limit,
+	}
+
+	tabWriter := printers.GetNewTabWriter(o.Out)
+
 	for {
-		cql, err := o.Client.ClusterQueues().List(ctx, metav1.ListOptions{
-			LabelSelector: o.LabelSelector,
-			FieldSelector: o.FieldSelector,
-			Limit:         o.clusterQueueListRequestLimit,
-			Continue:      continueToken,
-		})
+		noHeaders := totalCount > 0
+
+		list, err := o.Client.ClusterQueues().List(ctx, opts)
 		if err != nil {
 			return err
 		}
-		if len(cql.Items) == 0 {
-			return nil
-		}
 
-		o.applyActiveFilter(cql)
+		o.filterList(list)
+
+		totalCount += len(list.Items)
+
+		printer, err := o.ToPrinter(noHeaders)
+		if err != nil {
+			return err
+		}
 
-		if err := o.PrintObj(cql, o.Out); err != nil {
+		if err := printer.PrintObj(list, tabWriter); err != nil {
 			return err
 		}
 
-		if cql.Continue == "" {
+		if list.Continue != "" {
+			opts.Continue = list.Continue
+			continue
+		}
+
+		if totalCount == 0 {
+			fmt.Fprintln(o.ErrOut, "No resources found")
 			return nil
 		}
-		continueToken = cql.Continue
+
+		if err := tabWriter.Flush(); err != nil {
```

Comment:
what happens if the first 100 elements have rows with very few characters, and the next 100 characters have long values?
Would the output be misaligned?

Reply:
No, `TabWriter` will align.